### PR TITLE
fix: sockLen was being miscalculated when removing sockets

### PR DESCRIPTION
**Pull request by cixel**
Order of operations issue with + and ternary statements:

    console.log(0 + '' ? 'a' : 'b') // a

The code:

    freeLen + this.sockets[name] ? this.sockets[name].length : 0;

Is equivalent to:

    (freeLen + this.sockets[name]) ? this.sockets[name].length : 0;

When `this.sockets[name]` exists, it is an array, `(freeLen + this.sockets[name])` evaluates to a string, which evaluates to a string (true). When it does not, `(freeLen + this.sockets[name])` evaluates to `NaN` (false).

Either way, `freeLen` is ignored, which does not seem intended.

### CODE REVIEW

**Comment by tony-gutierrez** [comment]
This seems like a pretty important fix for potential bugs? Why no movement here?

**Comment by fengmk2** [comment]
can you add a test case for this fix?

**Comment by cixel** [comment]
Sorry for sleeping on this, I must've missed the notifications back in July. I took another look at this today to see how I could go about adding a test. 

I'll admit I'm not very familiar with the code; I found this issue doing static analysis with an AST parser on a project dependent on this one.

I made the following change to see if I could better understand the effects this issue has:
```javascript
  // [patch start]
  var freeLen = this.freeSockets[name] ? this.freeSockets[name].length : 0;
  var correctSockLen = freeLen + (this.sockets[name] ? this.sockets[name].length : 0);
  var sockLen = freeLen + this.sockets[name] ? this.sockets[name].length : 0;
  // [patch end]

  if (correctSockLen !== sockLen) {
	console.log(correctSockLen, sockLen, this.maxSockets);
  }

  if (this.requests[name] && this.requests[name].length && sockLen < this.maxSockets) {
    // ...
  }
```

During testing, there's never a situation where `correctSocklen !== sockLen` where `!(correctSockLen < this.maxSockets)`. This is the situation my patch would correct; however, I never see that, and I'm not sure how to cause that situation.

Any suggestions would be appreciated.

File: lib/_http_agent.js [hunk 1]
```diff
@@ -380,7 +380,7 @@ Agent.prototype.removeSocket = function removeSocket(s, options) {
 
   // [patch start]
   var freeLen = this.freeSockets[name] ? this.freeSockets[name].length : 0;
-  var sockLen = freeLen + this.sockets[name] ? this.sockets[name].length : 0;
```
Comment (fengmk2) [inline lib/_http_agent.js]: `freeLen + this.sockets[name] ? this.sockets[name].length : 0` is same as `freeLen + (this.sockets[name] ? this.sockets[name].length : 0)`

**Comment by fengmk2** [comment]
@cixel I see where is the bug now:

```
 node
> 1 + true ? 1 : 0
1
> 1 + (true ? 1 : 0)
2
```

**Comment by fengmk2** [comment]
3.5.2 released

@cixel Thanks.

**Comment by cixel** [comment]
Thank you! Sorry for the delayed follow-up.

**Pull request merged**